malwarecage/resources/config.py

```diff
@@ -167,14 +172,40 @@ def get(self, identifier):
 
     def _create_object(self, spec, parent, share_with, metakeys):
         try:
-            return Config.get_or_create(
-                spec.data["cfg"],
+            blobs = []
+            config = dict(spec.data["cfg"])
+            for first, second in config.items():
+                if isinstance(second, dict):
+                    for key in second:
+                        if key == "in-blob" and isinstance(second[key], dict):
+                            if not g.auth_user.has_rights(Capabilities.adding_blobs):
+                                abort(403)
+                            if not self.proper_text_blob_schema(second[key]):
+                                abort(400)
```

Comment:
Use `BlobCreateSpecSchema` and marshmallow instead of implementing own validation.